burrito-fg/src/lib.rs

```diff
@@ -57,6 +57,14 @@ impl X11_FG {
         godot_print!("Setting transient property failed");
         return false;
     }
+
+    #[export]
+    fn get_gw2_geometry(&self, _owner: &Node) -> (u16, u16){
+        let mut dpy = DisplayConnection::create(None, None).unwrap();
+        let window = get_window_from_name(&mut dpy, String::from("Guild Wars 2"));
```

Comment:
this has the issue that if guild wars didn't start yet, overlay will crash. 
 get window from name had issues. It didn't work on my pc last time :( the little updater window and the login window before the char select screen, both have the same window id / window name, so overlay will resize to them prematurely.

 you can use the x11_window_id_gw2 similar to the function that sets transient_for . this has the advantage, that we can be sure that gw2 has already started and is logged into character screen.

Reply:
I'm not saying that we shouldn't use id_from_window_name function because it didn't work on my pc. but because it was super hard to actually debug that kind of issue, so if other users get similar issues, debugging will be that much harder, compared to just using the xid that we get from mumble link reliably.

Reply:
Hmm yeah I did look at `x11_window_id_gw2` but on my machine it did not seem to have the right x11 id when I printed it out. That's why I didn't use it.

But sure let me take a look again, I might be missing something obvious. Will see if I can modify the function to take a x11 window id, and pass that in.

Reply:
I agree with @coderedart that we should try to use the xid value passed in similar to how `set_transient_for` works. Something to note is that the xid and pid are different values entirely. I would be surprised to learn that wine is setting a non-null incorrect xid value.

Because Burrito does not really do anything until the first time it receives a message from burrito link I believe we can just hold off on resizing until after we receive that first message. This would avoid crashing due to not having elements of that message already sent.

Reply:
To avoid starting at random sizes or too big like 1920x1080, maybe we can set to a default startup dimensions. I chose 800x600 as that seems to be a popular basic size. Then, when gw2 xid is available, burrito can resize to gw2 size.

FHD is very common, but due to taskbar (especially when it hides and comes back), burrito icon almost always is in a weird position. 800x600 can be more easier to see the icon on desktop .

Reply:
Seems reasonable to start smaller, then grow. We could even start with a small splash screen like window. I don't want to put the burden on this PR for that but we can open a new issue to design it further.